src/dmd/dmangle.d

```diff
@@ -225,13 +223,13 @@ public:
     {
         if (!t.isTypeBasic())
         {
-            auto p = cast(size_t*)dmd_aaGet(&types, cast(Key)t);
-            if (*p)
+            auto p = cast(void*)t in types;
+            if (p && *p)
             {
                 writeBackRef(buf.offset - *p);
                 return true;
             }
-            *p = buf.offset;
+            types[cast(void*)t] = buf.offset;
```

Comment:
Not too happy about the duplicate lookups here...

Reply:
Me neither. I suspect it can cause serious slowdowns.